e2e/pages/caseList.page.js

```diff
@@ -44,16 +44,17 @@ module.exports = {
   },
 
   async searchForCasesWithName(caseName, state='Any') {
-    this.setInitialSearchFields(state);
+    await this.setInitialSearchFields(state);
     // wait for our filters to load
     I.waitForVisible(this.fields.caseName, 30);
     I.fillField(this.fields.caseName, caseName);
     I.click(this.fields.search);
     await I.runAccessibilityTest();
   },
 
-  setInitialSearchFields(state='Any') {
+  async setInitialSearchFields(state='Any') {
     // wait for initial filters to load
+    await I.waitForSpinnerToFinish();
```

Comment:
Does the spinning not occur after the search? Or does it do spinning as well whilst populating the search fields as well?

Reply:
The spinning that I saw interfering with the tests wasn't when it was waiting for a case to load, but when the test was using a select dropdown which was already on that page _behind_ a loading spinner. If the test step was waiting for a case to load it seemed to work, I'm thinking because the spinner disappears once the search is complete and thats when the searched case is present.
But it wouldn't be much effort to go with you idea of using a hook to check it's presence before any step

Reply:
Ahh it loads the case list when you first go to the page as well using the default filters... ignore this 😄

Reply:
The hook might be a nicer idea, that way if they introduce it anywhere else then we can automatically wait for it

Reply:
I guess we would need something that is constant across all pages that we can use to check, maybe the case list/create case/sign out links in the banner?